genisys/modules/os_file_download.py

```diff
@@ -0,0 +1,35 @@
+'''
+This module will handle the downloading of all of the files necessary for the PXE booting of a client machine. 
+
+NOTES:
+Downloading and storing all of the various files needed for boot:
+syslinux.pxe / whatever
+post-boot ISO/debian.whatever files
+Need to download and install these files to the correct directories
+Look into netboot.xyz
+
+- vmlinuz (Kernel) & initrd/initramfs (Initial Ram Disk)
+- pxelinux.cfg 
+- pxelinux.0 or GRUB (Depends on OS/Version)
+
+Link to Debian files (amd64): https://deb.debian.org/debian/dists/bookworm/main/installer-amd64/current/images/ 
+'''
+from genisys.modules import base
+from pathlib import Path
+from typing_extensions import Self, Union, List
+
+class OSDownload(base.Module):
+    DEBIAN_TAR_FILE_LINK = "https://deb.debian.org/debian/dists/bookworm/main/installer-amd64/current/images/netboot/netboot.tar.gz" 
```

Comment:
Just for future reference: we may want to update the configuration file to use `version: bookworm` and add a new `target-architecture:` field. Then we can generate this like with something like `f"https://deb.debian.org/debian/dists/{version}/main/installer-{arch}/current/images/netboot/netboot.tar.gz"`

Reply:
Addressed in commit [6909c4d](https://github.com/gingrm02/genisys/pull/16/commits/6909c4d1ea6cd3c017cf11aa0752138d945c6416)